src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh

```diff
@@ -2,9 +2,23 @@
 #   Copyright (c) Microsoft Corporation. All rights reserved.
 #   Licensed under the MIT License. See License.txt in the project root for license information.
 # ---------------------------------------------------------------------------------------------
-
 autoload -Uz add-zsh-hook
 
+. ~/.zshenv
+if [ -z "$VSCODE_SHELL_LOGIN" ]; then
+	if [ -f ~/.zprofile ]; then
+		. ~/.zprofile
+	fi
+fi
+. ~/.zshrc
+if [ -z "$VSCODE_SHELL_LOGIN" ]; then
+	if [ -f ~/.zlogin ]; then
+		. ~/.zlogin
+		VSCODE_SHELL_LOGIN=""
+	fi
+fi
+
+
```

Comment:
I think we need to unset ZDOTDIR at this point to get this to work:

> When a login shell exits, the files $ZDOTDIR/.zlogout and then /etc/zlogout are read.

https://linux.die.net/man/1/zsh